gen/crocs_user_export

```diff
@@ -8,36 +8,35 @@ use JSON::XS;
 
 our $SERVICE_NAME = "crocs_user_export";
 our $PROTOCOL_VERSION = "3.0.0";
-my $SCRIPT_VERSION = "3.0.0";
+my $SCRIPT_VERSION = "3.0.1";
 
 perunServicesInit::init;
 my $DIRECTORY = perunServicesInit::getDirectory;
-my $data = perunServicesInit::getHierarchicalData;
+my $data = perunServicesInit::getHashedHierarchicalData;
 
 #Constants
 our $A_USER_FIRSTNAME;            *A_USER_FIRSTNAME =        \'urn:perun:user:attribute-def:core:firstName';
 our $A_USER_LASTNAME;             *A_USER_LASTNAME =         \'urn:perun:user:attribute-def:core:lastName';
 our $A_USER_LOGIN;                *A_USER_LOGIN=             \'urn:perun:user:attribute-def:def:login-namespace:mu';
 
-my %attributesByLogin;
+my $attributesByLogin;
 
-my @resourcesData = $data->getChildElements;
-foreach my $rData (@resourcesData) {
-	my @membersData = $rData->getChildElements;
-	foreach my $mData (@membersData) {
-		my %memberAttributes = attributesToHash $mData->getAttributes;
-		$attributesByLogin{$memberAttributes{$A_USER_LOGIN}} = \%memberAttributes;
-	}
+foreach my $memberId ($data->getMemberIdsForFacility()) {
+	my $login = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_LOGIN );
+
+	$attributesByLogin->{$login}->{$A_USER_FIRSTNAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_FIRSTNAME );
+	$attributesByLogin->{$login}->{$A_USER_LASTNAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_LASTNAME );
 }
 
 
 my @users;
-for my $login (sort keys %attributesByLogin) {
-	my $values = $attributesByLogin{$login};
+for my $login (sort keys %{$attributesByLogin}) {
+	my $values = $attributesByLogin->{$login};
 	my $user = {};
 	$user->{"firstname"} = $values->{$A_USER_FIRSTNAME} || "";
 	$user->{"lastname"}  = $values->{$A_USER_LASTNAME} || "";
-	$user->{"UCO"}       = $values->{$A_USER_LOGIN};
+	$user->{"UCO"}       = $values;
```

Comment:
You should print login (aka UCO) here and not the whole `$values` structure. Previously it printed object like:
```json
{
      "firstname" : "GivenName",
      "UCO" : "123456",
      "lastname" : "Surname"
}
```
Now it prints:
```json
{
      "lastname" : "Surname",
      "UCO" : {
         "urn:perun:user:attribute-def:core:firstName" : "GivenName",
         "urn:perun:user:attribute-def:core:lastName" : "Surname"
      },
      "firstname" : "GivenName"
}
```

Reply:
Also please use `canonical` while printing JSON, so that objects have sorted keys and results are comparable between the runs, like: 
```perl
JSON::XS->new->utf8->pretty->canonical->encode(\@users);
```